typescript/src/codeActions/custom/renameToTypeParameter.ts

```diff
@@ -0,0 +1,53 @@
+import { pipe, groupBy, map } from 'lodash/fp'
+import { CodeAction } from '../getCodeActions'
+import extractType from '../../utils/extractType'
+
+export default {
+    id: 'renameToTypeParameter',
+    name: 'Rename to Type Parameter',
+    kind: 'refactor.rewrite.renameToTypeParameter',
+    tryToApply(sourceFile, position, range, node, formatOptions, languageService) {
+        if (!node || !position) return
+        if (!ts.isIdentifier(node) || !ts.isParameter(node.parent) || !ts.isFunctionLike(node.parent.parent)) return
+
+        const functionParameters = node.parent.parent.parameters
+        const parameterIndex = functionParameters.indexOf(node.parent)
+
+        const functionDecl = node.parent.parent.parent
+        const typeChecker = languageService.getProgram()!.getTypeChecker()
+        const type = extractType(typeChecker, functionDecl)
+
+        const typeDecl = type.getSymbol()?.declarations?.[0]
+        if (!typeDecl || !ts.isFunctionLike(typeDecl)) return
+        const paramName = node.getText()
+        const typeParamName = typeDecl.parameters[parameterIndex]!.name.getText()
+        if (paramName === typeParamName) return
+        if (!formatOptions) return true
+
+        const renameLocations = languageService.findRenameLocations(sourceFile.fileName, position, false, false, {
+            providePrefixAndSuffixTextForRename: false,
+        })
+        if (!renameLocations) return
+
+        const extractFileName = ({ fileName }: ts.RenameLocation) => fileName
+        const edits = pipe(
+            groupBy(extractFileName),
+            Object.entries,
+            map(
+                ([fileName, changes]): ts.FileTextChanges => ({
+                    fileName,
+                    textChanges: changes.map(
+                        ({ textSpan }): ts.TextChange => ({
+                            newText: typeParamName,
+                            span: textSpan,
+                        }),
+                    ),
+                }),
+            ),
+        )(renameLocations)
```

Comment:
I could use your implementation, but this one looks more readable to me :). I can revert if you don't like  it
https://github.com/zardoy/typescript-vscode-plugins/blob/240f2e1cc89f871d2442d5b28270b38dcd51ab2f/typescript/src/decorateEditsForFileRename.ts#L21-L31